Gateway pods fail to start when namespace doesn't exist (#397)
hashicorp/consul-api-gateway: .changelog/397.txt

```release-note:bug

> When a gateway is created in a namespace that doesn't exist in consul, and namespace mirrioring is enabled, the namespace will be created in consul. 

Reviewer: ```suggestion
When a gateway is created in a namespace that doesn't exist in Consul and namespace mirroring is enabled, create the namespace in Consul. 
```
